src/cargo/util/restricted_names.rs

```diff
@@ -87,23 +87,22 @@ pub fn validate_package_name(name: &str, what: &str, help: &str) -> CargoResult<
 pub fn sanitize_package_name(name: &str, placeholder: char) -> String {
     let mut slug = String::new();
     let mut chars = name.chars();
-    if let Some(ch) = chars.next() {
-        if ch.is_digit(10) {
-            slug.push(placeholder);
-            slug.push(ch);
-        } else if unicode_xid::UnicodeXID::is_xid_start(ch) || ch == '_' {
+    while let Some(ch) = chars.next() {
+        if (unicode_xid::UnicodeXID::is_xid_start(ch) || ch == '_') && !ch.is_digit(10) {
             slug.push(ch);
-        } else {
-            slug.push(placeholder);
+            break;
         }
     }
-    for ch in chars {
+    while let Some(ch) = chars.next() {
         if unicode_xid::UnicodeXID::is_xid_continue(ch) || ch == '-' {
             slug.push(ch);
         } else {
             slug.push(placeholder);
         }
     }
+    if slug.is_empty() {
+        slug.push_str("package");
```

Comment:
I might prefer `script` but never mind it's not a big deal.

Reply:
I'm avoiding "script" in user-facing contexts because I feel it can be misleading.

Reply:
I'm avoiding "script" in user-facing contexts because I feel it can be misleading.